### PR TITLE
refactor: move constants into config and utils, resolves #19

### LINKED ISSUE #19: 🚀 Feature: Store constant values in a file for reuse
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/philly-js-club/philly-js-club-website-remix/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

Currently, no data file contains the fundamental constant values repeated across the website.
The most concerning example is the site title (Philly JS Club) which is currently repeated for every page title and will require a global replacement in case of future changes (not that I believe it should be changed ;)).

It will be handy to have some constants repeated across the website stored somewhere; We might even benefit from some utility functions for common cases like constructing the site title.

### Additional Info

It would be neat to have something like:
```ts
import site from 'config/site';

export const meta: V2_MetaFunction = () => {
  return [{ title: `Some Page | ${site.title}` }];
}
```

Or even:
```ts
import { constructSiteTitle } from 'utils/site';

export const meta: V2_MetaFunction = () => {
  return [{ title: constructSiteTitle("Some page") }];
}
```

I'd be glad to hear your thoughts, it's the most raw idea and I expect some brainstorming around it.

### PR DESCRIPTION
**Pull request by RNR1**
## PR Checklist

- [x] Addresses an existing open issue: fixes #19
- [x] That issue was marked as [`status: accepting prs`](https://github.com/philly-js-club/philly-js-club-website-remix/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/philly-js-club/philly-js-club-website-remix/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

I went ahead and moved some of the constant values in the project:
- site constants go into `config/site.ts` 
- region constants go into `config/region.ts`

I also moved utils into `utils/common.ts`; this includes a new function to construct the page title so that we will define the page title once.

Last, I moved the team data from `routes/about.tsx` into `data/team.json`

Sorry in advance for the amount of file changes 🤓

### CODE REVIEW

File: app/config/region.ts
Comment (JoshuaKGoldberg) [inline app/config/region.ts]: [Style] How certain are you that you prefer `export default` in `.ts`/`.tsx` files? 😄 I'm pretty -1 on it in general, for the reasons in the first ~half of https://basarat.gitbook.io/typescript/main-1/defaultisbad.

What are your thoughts on switching to `export const site = ...`? _(and maybe combining into `app/config.ts`, since the files are pretty small as-is?)

Alternately, I also don't see a strong delineation between `app/config/*.ts` and `app/data/*.json`. Maybe these could all be `.json` files, ironically which would then have to be `default`-imported? 

FWIW my personal instinct is to go with a style that results in imports like `import { region } from "~/config"`. But I'm definitely not the only voice here that should be heard.

File: app/routes/about.tsx
Comment (JoshuaKGoldberg) [inline app/routes/about.tsx]: [Refactor] Think we should also put `organizersEmeritum` into the data place, along with `team`?

File: app/utils/common.ts
Comment (JoshuaKGoldberg) [inline app/utils/common.ts:15]: [Praise] Yeah I really like having this unified function 😄 +1

File: app/utils/common.ts
Comment (JoshuaKGoldberg) [inline app/utils/common.ts]: [Docs] Type annotations such as `{string}` aren't generally necessary in JSDoc params. TypeScript does that for us already. So we can omit a bit of them:

```suggestion
 * @param pageTitle - The current page title
```

I've actually been working on this recently in linter-land and plan on bringing in some of the lint rules to this site eventually 😄 

* https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues/303
* https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues/371
* https://github.com/gajus/eslint-plugin-jsdoc/pull/1007

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
This is great! Definitely a positive step forward. Love to see constants getting shared as variables.

None of my comments are particularly blocking. It's all small stuff I just want to chat about before merging. Cheers! 🙌

File: app/config/region.ts
Comment (RNR1) [reply app/config/region.ts]: The link you provided had strong opinions I can understand and respect.
My take is that default exports/imports are a good way to define the primary purpose of a module, so it's my go-to for files with a single goal. 
It helps me be focused when structuring different app models, directory structures, etc...
This is why I also chose a design of single-focused files.

On the one hand, when default exports are handled correctly, they can maintain very focused modules toward the app.
On the other hand, as the article you provided explains, it can go wrong when they are not handled in a specific manner.

I have two suggestions in mind:
1. Take your approach for a single config file - it can keep our imports aligned; The main drawback is that a single config file can grow over-limit, though it may not be very likely for this app.
2. Add a "barrel" `index.ts` file for exporting each module in the config directory so that each module will be exported as named export.

I think it might be more likely to take the first approach honestly, but we can have a discussion around it

File: app/routes/about.tsx
Comment (RNR1) [reply app/routes/about.tsx]: Maybe separate this file into "organizers" and "organizersEmeritum"

File: app/config/region.ts
Comment (RNR1) [reply app/config/region.ts]: The last thing I missed regarding moving to a `.json` config file over `.ts`: 
The main benefit is that in .ts I can use `as const` to get a better IntelliSense for consts, which is neat :)

image

File: app/routes/about.tsx
Comment (RNR1) [reply app/routes/about.tsx]: Updated, also now using a loader to make typescript and remix happy :)

File: app/utils/common.ts
Comment (RNR1) [reply app/utils/common.ts]: Updated

File: app/config/region.ts
Comment (JoshuaKGoldberg) [reply app/config/region.ts]: > single config file

Yeah that'd be my preference. Barrel (`index.ts`) re-exports are nice when you have a lot, but we only have a couple files. Seems premature to me.

> `as const`

+1 that is a great benefit! 😄

File: app/utils/common.ts
Comment (JoshuaKGoldberg) [inline app/utils/common.ts]: [Nit] I don't think either of these give much benefit.
```suggestion
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Sweet! I'm going to hold off merging to let other folks to weigh in on the stylistic conversation. But know that if it was just us I'd be very happy to merge 😄 thanks!

**Comment by RNR1** [comment]
Great, just applied the most recent suggestions :)

**Comment by tjwds** [comment]
LGTM — thanks so much @RNR1, and thanks for the review, @JoshuaKGoldberg!

**Pull request merged**
